client/Components/map.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useRef, useCallback, useContext, useEffect } from 'react';
+import EggIcon from '../UI/egg-icon';
+import MapGL, { Marker, GeolocateControl } from 'react-map-gl';
+import Geocoder from 'react-map-gl-geocoder';
+import Navbar from './navbar';
+import CreateEgg from './createEggModal';
+import { UserContext } from '../Context/userContext';
+
+const MAPBOX_TOKEN = 'pk.eyJ1IjoibXBvd2VyczExMTMiLCJhIjoiY2t4Mmo3MTl1MWlyNzJucDJ1d2xtZXA4aiJ9.w4OiI4e8mDRubQC-0vJs3g';
```

Comment:
Typically we don't hard-code keys into the front end. Instead, use webpack's environment plugin to read the value from your `.env` file.

https://webpack.js.org/plugins/environment-plugin/

Here is an example
https://github.com/barbara-liao/wanderlist/blob/master/webpack.config.js#L2
https://github.com/barbara-liao/wanderlist/blob/master/webpack.config.js#L49-L51